src/actions/manager.tsx

```diff
@@ -4,13 +4,14 @@ import {
   ActionsManagerInterface,
   UpdaterFn,
   ActionFilterFn,
+  ActionName,
 } from "./types";
 import { ExcalidrawElement } from "../element/types";
 import { AppState } from "../types";
 import { t } from "../i18n";
 
 export class ActionManager implements ActionsManagerInterface {
-  actions: { [keyProp: string]: Action } = {};
+  actions = {} as ActionsManagerInterface["actions"];
```

Comment:
We need to cast here if we want to retain the non-optionality of the values. While this isn't type-safe, AFAIK it doesn't reduce overall codebase type-safety because previously the index signature allowed for the same thing, except it was wider. Now we are at least guaranteed we don't mistype the action name.

Reply:
Records work better here

Reply:
`actions = {} as Record<ActionName, Action>;`

Reply:
I'm new to Records, what's the benefit?

Reply:
Just checked the `Record` implem. It's the same thing :smile_cat:

Reply:
Yea it is. just more readable